packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_options.dart

```diff
@@ -0,0 +1,152 @@
+        bundleID = map['bundleID'],
+        clientID = map['clientID'],
+        trackingID = map['trackingID'],
+        gcmSenderID = map['GCMSenderID'],
+        projectID = map['projectID'],
+        androidClientID = map['androidClientID'],
+        googleAppID = map['googleAppID'],
+        databaseURL = map['databaseURL'],
+        deepLinkURLScheme = map['deepLinkURLScheme'],
+        storageBucket = map['storageBucket'] {
+    assert(googleAppID != null);
+  }
+
+  /// An API key used for authenticating requests from your app, e.g.
+  /// "AIzaSyDdVgKwhZl0sTTTLZ7iTmt1r3N2cJLnaDk", used to identify your app to
+  /// Google servers.
+  ///
+  /// This property is required on Android.
+  final String apiKey;
+
+  /// The iOS bundle ID for the application. Defaults to
+  /// `[[NSBundle mainBundle] bundleID]` when not set manually or in a plist.
+  ///
+  /// This property is used on iOS only.
+  final String bundleID;
+
+  /// The OAuth2 client ID for iOS application used to authenticate Google
+  /// users, for example "12345.apps.googleusercontent.com", used for signing in
+  /// with Google.
+  ///
+  /// This property is used on iOS only.
+  final String clientID;
+
+  /// The tracking ID for Google Analytics, e.g. "UA-12345678-1", used to
+  /// configure Google Analytics.
+  ///
+  /// This property is used on iOS only.
+  final String trackingID;
+
+  /// The Project Number from the Google Developer’s console, for example
+  /// "012345678901", used to configure Google Cloud Messaging.
+  ///
+  /// This property is required on iOS.
+  final String gcmSenderID;
+
+  /// The Project ID from the Firebase console, for example "abc-xyz-123."
+  final String projectID;
+
+  /// The Android client ID, for example "12345.apps.googleusercontent.com."
+  ///
+  /// This property is used on iOS only.
+  final String androidClientID;
+
+  /// The Google App ID that is used to uniquely identify an instance of an app.
+  ///
+  /// This property cannot be `null`.
+  final String googleAppID;
+
+  /// The database root URL, e.g. "http://abc-xyz-123.firebaseio.com."
+  ///
+  /// This property should be set for apps that use Firebase Database.
+  final String databaseURL;
+
+  /// The URL scheme used to set up Durable Deep Link service.
+  ///
+  /// This property is used on iOS only.
+  final String deepLinkURLScheme;
+
+  /// The Google Cloud Storage bucket name, e.g.
+  /// "abc-xyz-123.storage.firebase.com."
+  final String storageBucket;
+
+  Map<String, String> get asMap {
+    return <String, String>{
+      'APIKey': apiKey,
+      'bundleID': bundleID,
+      'clientID': clientID,
+      'trackingID': trackingID,
+      'GCMSenderID': gcmSenderID,
+      'projectID': projectID,
+      'androidClientID': androidClientID,
+      'googleAppID': googleAppID,
+      'databaseURL': databaseURL,
+      'deepLinkURLScheme': deepLinkURLScheme,
+      'storageBucket': storageBucket,
+    };
+  }
+
+  @override
+  bool operator ==(dynamic other) {
```

Comment:
We need something like [lombok](https://projectlombok.org/features/EqualsAndHashCode) for these POJOs, I've found myself writing stuff similar to this lately and it's a chore.

Reply:
There is [`built_value`](https://medium.com/dartlang/darts-built-value-for-immutable-object-models-83e2497922d4#.48dyezxcl) but I'm not sure how good the support is in Flutter for code generation via `build_runner`

Reply:
Very similar to what the Dart implementation of [lombok](https://pub.dev/packages/lombok) is doing! Thanks for the link!